keyrings/alt/file_base.py

```diff
@@ -61,6 +97,7 @@ def get_password(self, service, username):
         """
         service = escape_for_ini(service)
         username = escape_for_ini(username)
+        assoc = (service + '\0' + username).encode()
```

Comment:
Because I wanted to be sure to get exactly the service and username bytestrings (okay, due to value escaping, this is tamper resistant already, hmm..) 

assoc is a sensible value, that needs to be reproduced identically during decoding (or resulting in MAC failures). When using a dict, it will have to be an ordered one, but I avoided to act too subtle in this area. A tuple would work, but that has to be serialized to a byte string anyway.